slurminade/dispatcher.py

```diff
@@ -130,20 +135,27 @@ def __init__(self):
         self.calls = []
         self.sbatches = []
         self.sruns = []
+        self.max_arg_length = DEFAULT_MAX_ARG_LENGTH
 
     def _dispatch(
         self, funcs: typing.Iterable[FunctionCall], options: SlurmOptions
     ) -> int:
         dispatch_guard()
         funcs = list(funcs)
-        print(
-            f"{sys.executable} -m slurminade.execute"
-            f" {shlex.quote(get_entry_point())}"
-            f" {shlex.quote(json.dumps([f.to_json() for f in funcs]))}"
-        )
+        command = create_slurminade_command(funcs, self.max_arg_length)
+        print(command)
         self.calls.append(funcs)
+        self._cleanup(command)
         return -1
 
+    def _cleanup(self, command):
+        args = shlex.split(command)
+        if args[-2] != "temp":
```

Comment:
This is kind  of ugly. It requires the command to keep its syntax. This is a hidden dependency that could blow up at some point. At least make it some function that is collocated with the command creation.